repository/error/error.go

```diff
@@ -0,0 +1,54 @@
+package error
+
+import (
+	"errors"
+
+	svcerr "release-manager/service/error"
+
+	postgrestgo "github.com/nedpals/postgrest-go/pkg"
+	"github.com/nedpals/supabase-go"
+)
+
+var (
+	PostgresResourceNotFoundCode = "PGRST116"
+)
+
+var (
+	ErrMultipleOrNoRecordsReturned = errors.New("supabase: one record expected, multiple (or no) rows returned")
+)
+
+func ToSvcErrFromAuth(err error) error {
+	var supabaseErr *supabase.ErrorResponse
+	if errors.As(err, &supabaseErr) {
+		switch supabaseErr.Code {
+		case 401, 404:
+			return svcerr.NewAuthErr(err.Error())
+		}
+	}
+
+	return err
+}
+
+func ToSvcErrFromDB(err error) error {
+	var postgreErr *postgrestgo.RequestError
+	if errors.As(err, &postgreErr) {
+		switch postgreErr.Code {
+		case PostgresResourceNotFoundCode:
+			return svcerr.NewResourceNotFoundErr(err.Error())
+		}
+	}
+
+	return err
+}
+
+func ToSvceErrFromAdmin(err error) error {
+	var supabaseErr *supabase.ErrorResponse
+	if errors.As(err, &supabaseErr) {
+		switch supabaseErr.Code {
+		case 404:
```

Comment:
In Admin (which I use only for DELETE /admin/users/{id}) error 404 means that user was not found. I dont want to map any other error to service errors.